src/lib.rs

```diff
@@ -332,14 +332,21 @@ impl State {
             }
 
             // Timeout is contemplated in this join_all because of the controller spawning tasks.
-            let tasks = join_all(handles).await;
+            let stream_iter = iter(futures);
+            let mut buffered_iter = stream_iter.buffer_unordered(self.config.user_act_throughput);
 
             // Report errors for each task that could not complete
-            for task in tasks {
-                if task.is_err() {
-                    tx.send(Event::SendMessageCancelledError)
-                        .await
-                        .expect("channel shouldn't be closed");
+            while let Some(task) = buffered_iter.next().await {
+                match task {
+                    Ok(None) => {
+                        tx.send(Event::SendMessageCancelledError)
+                            .await
+                            .expect("channel shouldn't be closed");
+                    }
+                    Err(_) => {
+                        log::info!("JoinError");
```

Comment:
Can we have these? What would be the reason?

Reply:
Yes, I think we can but unlikely. `JoinError` means the task was canceled or panicked.

Reply:
True that! Let's track this event in that case